src/charm.py

```diff
@@ -36,6 +37,10 @@ def __init__(self, *args):
             self.framework.observe(event, self.main)
 
     def main(self, event):
+        if not self.unit.is_leader():
+            self.model.unit.status = WaitingStatus("Waiting for Leadership")
+            exit()
+
```

Comment:
I would prefer if this is done the same way as [here](https://github.com/canonical/minio-operator/blob/54adfe3173aedde843b42c49244d91663258c95f/src/charm.py#L44)